homeassistant/components/google/__init__.py

```diff
@@ -188,7 +188,12 @@ def do_authentication(hass, hass_config, config):
 
     def step2_exchange(now):
         """Keep trying to validate the user_code until it expires."""
-        if now >= dt.as_local(dev_flow.user_code_expiry):
+
+        # oauth.step1_get_device_and_user_codes() returns a dev_flow.user_code_expiry
+        # timestamp that is in UTC, but with tzinfo == None.
+        user_code_expiry = dev_flow.user_code_expiry.replace(tzinfo=timezone.utc)
```

Comment:
Is this good as is, or should it be conditionalized and only done if `tzinfo is None`?

Reply:
If we know that this is UTC time I think it's ok like this.